.github/workflows/docker_test.yml

```diff
@@ -27,15 +43,23 @@ jobs:
     name: Test docker images
     steps:
       - uses: actions/checkout@v4
+      - name: "Setup Environment"
+        uses: ./.github/actions/setup-uv
+      - name: Install the project
+        run: uv sync --dev
+      - name: Debug Inputs
```

Comment:
I don't dislike having the echos. Though, you could just move them to the `Build Image` step before docker build is run.